lib/tokens/IamTokenManager.js

```diff
@@ -0,0 +1,99 @@
+// Copyright © 2019 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+'use strict';
+
+const a = require('async');
+const debug = require('debug')('cloudant:tokens:iamtokenmanager');
+const TokenManager = require('./TokenManager');
+
+class IAMTokenManager extends TokenManager {
+  constructor(client, jar, sessionUrl, iamTokenUrl, iamApiKey, iamClientId, iamClientSecret) {
+    super(client, jar, sessionUrl);
+
+    this._iamTokenUrl = iamTokenUrl;
+    this._iamApiKey = iamApiKey;
+    this._iamClientId = iamClientId;
+    this._iamClientSecret = iamClientSecret;
+  }
+
+  _getToken(done) {
+    var self = this;
+
+    debug('Making IAM session request.');
+    let accessToken;
+    a.series([
+      (callback) => {
+        let accessTokenAuth;
+        if (self._iamClientId && self._iamClientSecret) {
+          accessTokenAuth = { user: self._iamClientId, pass: self._iamClientSecret };
+        }
+        debug('Getting access token.');
+        self._client({
+          url: self._iamTokenUrl,
+          method: 'POST',
+          auth: accessTokenAuth,
+          headers: { 'Accepts': 'application/json' },
```

Comment:
Does this need to specify the `Content-Type` of `application/x-www-form-urlencoded` as well?

Reply:
Nope. Request does this for us.